tests/hypothesis_strategies/constraints.py

```diff
@@ -0,0 +1,149 @@
+"""Hypothesis strategies for constraints."""
+
+from functools import partial
+from typing import Any, List, Optional, Type, Union
+
+import hypothesis.strategies as st
+
+from baybe.constraints.conditions import (
+    SubSelectionCondition,
+    ThresholdCondition,
+    _valid_logic_combiners,
+)
+from baybe.constraints.continuous import (
+    ContinuousLinearEqualityConstraint,
+    ContinuousLinearInequalityConstraint,
+)
+from baybe.constraints.discrete import (
+    DiscreteExcludeConstraint,
+    DiscreteLinkedParametersConstraint,
+    DiscreteNoLabelDuplicatesConstraint,
+    DiscreteProductConstraint,
+    DiscreteSumConstraint,
+)
+from baybe.parameters.base import DiscreteParameter
+from baybe.parameters.numerical import NumericalDiscreteParameter
+
+
+def sub_selection_conditions(superset: Optional[List[Any]] = None):
+    """Generate :class:`baybe.constraints.conditions.SubSelectionCondition`."""
+    if superset is None:
+        element_strategy = st.text()
+    else:
+        element_strategy = st.sampled_from(superset)
+    return st.builds(SubSelectionCondition, st.lists(element_strategy, unique=True))
+
+
+def threshold_conditions():
+    """Generate :class:`baybe.constraints.conditions.ThresholdCondition`."""
+    return st.builds(
+        ThresholdCondition, threshold=st.floats(allow_infinity=False, allow_nan=False)
+    )
+
+
+@st.composite
+def discrete_excludes_constraints(
+    draw: st.DrawFn, parameters: Optional[List[DiscreteParameter]] = None
+):
+    """Generate :class:`baybe.constraints.discrete.DiscreteExcludeConstraint`."""
+    if parameters is None:
+        parameter_names = draw(st.lists(st.text(), unique=True, min_size=1))
+        conditions = draw(
+            st.lists(
+                st.one_of(sub_selection_conditions(), threshold_conditions()),
+                min_size=len(parameter_names),
```

Comment:
Why is `min_size=max_size`?

Reply:
This is just the hypothesis way of saying "give me a list of exactly that size". There might be smarter / more efficient ways to create the objects, but I also always need to spend quite a while thinking until I get what I want. Suggestions welcome 😄